ESP8266/ESP8266.cpp

```diff
@@ -279,28 +287,24 @@ bool ESP8266::close(int id)
 
 void ESP8266::setTimeout(uint32_t timeout_ms)
 {
-    _parser.setTimeout(timeout_ms);
+    _parser->setTimeout(timeout_ms);
 }
 
-bool ESP8266::readable()
-{
-    return _serial.readable();
```

Comment:
poll() have a dummy implementation in FileHandle , the derived class should implement that. mbed_poll() on the other hand which is exposed by the FileHandle header file is blocking in nature.
I did not find any piece of code using these APIs that's why I removed it. You can implement poll() if necessary .. Or if you have something in your mind, go ahead and share with me